### DISCUSSION TITLE
Adding banner to compiled JS

### DISCUSSION THREAD

**Discussion by yoavniran** [remix-run/remix#5736] [Q&A]
I'd like to add the current build version to the JS sent to the browser 
Typically I'd do that by adding a "banner" comment to the top of the (chunk) file.
As you could do with webpack or rollup.

I haven't been able to find any mention of something like this for Remix. Is it possible?

Thanks!

**Comment by sergiodxa** [comment]
There's no built-in way, you could run a postbuild script to add it, but remember that if a file didn't changed the hash name doesn't change, so you could run `remix build` and have a file with the same file name as the previous build, if you add the build version to the file it will be the same file with the different build versions.

**Comment by yoavniran** [comment]
thanks @sergiodxa 
I dont like the approach of modifying the files outside the build flow... 😞 
Would have been much better to have access to the flow and/or its configuration to support this use-case.

I ended up going with a different approach - following the [tutorial on how to inject env variables to the client](https://egghead.io/lessons/remix-expose-specific-environmental-variables-to-the-client-using-remix). I now pass the VERSION as an env variable from my docker build (receiving it as arg from the GH Flow) and then adding it as a meta-tag to the HTML head. 

It makes it easy to inspect when needed. 🤷
